### PR TITLE
Withdraw unsold tokens

### PR DESCRIPTION
**Pull request by alex-miller-0**
Migration of #2. This allows the `owner` to withdraw all unsold tokens to the `wallet` after the `endBlock` is reached (i.e. after the sale ends).

### CODE REVIEW

File: contracts/Sale.sol [hunk 1]
```diff
@@ -216,6 +222,15 @@ contract Sale {
         owner = _newOwner;
     }
 
+    function withdrawRemainder()
+        onlyOwner
+        saleStarted
```
Comment (skmgoldin) [inline contracts/Sale.sol]: This seems unnecessary, no?

**Comment by skmgoldin** [comment]
I think you can probably squash some commits here? Should have two: "Add tests for withdrawRemainder" and "implement withdrawRemainder" or something. Just to keep the commit history clean.

File: contracts/Sale.sol (hunk 1 above)
Comment (alex-miller-0) [reply contracts/Sale.sol]: The reason I put that was to ward off any funny business of the `endBlock` being earlier than the start block. I suppose there would be no money if that's the case though :P

File: contracts/Sale.sol [hunk 2]
```diff
@@ -216,6 +222,15 @@ contract Sale {
...
+        saleEnded
+    {
+        uint remainder = token.balanceOf(this);
+        token.transfer(wallet, remainder);
```
Comment (skmgoldin) [inline contracts/Sale.sol]: Maybe this should take an argument for the transfer address. If a token seller uses, like, a Coinbase address for their wallet (they shouldn't, but...) and withdraws to it they'll lose their tokens.

Comment (skmgoldin) [reply contracts/Sale.sol]: Open to discussion on this though.

Comment (GNSPS) [reply contracts/Sale.sol]: I was thinking of that, but having still another wallet address in the config parameters is also not optimal...

Comment (skmgoldin) [reply contracts/Sale.sol]: You (the owner) would just pass it in as an argument to _this function_. No need for it in the constructor methinks.

File: test/saleB.js [hunk 3]
```diff
@@ -67,7 +67,7 @@ contract('Sale', (accounts) => {
 
   describe('Sale ends before all tokens are sold', () => {
     it('should stop selling tokens after the endBlock', async () => {
-      await forceMine(saleConf.endBlock.add(new BN('1', 10)));
+      await forceMine(saleConf.endBlock.add(new BN('2', 10)));
```
Comment (skmgoldin) [inline test/saleB.js]: Why this change?

File: contracts/Sale.sol (hunk 2 above)
Comment (GNSPS) [reply contracts/Sale.sol]: Ahhh, sure! 👍

File: test/saleB.js [hunk 4]
```diff
@@ -81,4 +81,21 @@ contract('Sale', (accounts) => {
...
     });
   });
+
+  it('should allow the owner to withdraw all unsold tokens', async () => {
+    await forceMine(saleConf.endBlock.add(new BN('2', 10)));
+
+    const sale = await Sale.deployed();
+    const startingBalanceSale = await getTokenBalanceOf(sale.address);
+    const owner = await sale.owner.call();
+    const receipt = await sale.withdrawRemainder({ from: owner });
```
Comment (skmgoldin) [inline test/saleB.js]: Can remove the assignment `const receipt =` since you don't use it.

File: test/saleB.js [hunk 5]
```diff
@@ -81,4 +81,21 @@ contract('Sale', (accounts) => {
...
+    const endingBalanceSale = await getTokenBalanceOf(sale.address);
+    const wallet = await sale.wallet.call();
+    const endingBalanceWallet = await getTokenBalanceOf(wallet);
+
+    const errMsg1 = 'Sale does not have 0 tokens, but should.';
+    const errMsg2 = 'Wallet did not receive withdrawn tokens from sale.';
+    assert.equal(endingBalanceSale.toString(10), 0, errMsg1);
```
Comment (skmgoldin) [inline test/saleB.js]: Use strictEqual.

`assert.strictEqual(endingBalanceSale.toString(10), '0', errMsg1);`

(We should do a separate PR to take out all the instances of `assert.equal` in the tests.

File: test/saleB.js (hunk 3 above)
Comment (alex-miller-0) [reply test/saleB.js]: Reverted

File: test/saleB.js (hunk 4 above)
Comment (alex-miller-0) [reply test/saleB.js]: Removed

File: test/saleB.js (hunk 5 above)
Comment (alex-miller-0) [reply test/saleB.js]: Fixed

**Comment by GNSPS** [comment]
LGTM. 👍 @skmgoldin I think we can merge without squashing commits, right?

**Comment by alex-miller-0** [comment]
Closing this in favor of a branch with a cleaner commit history.
